ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,284 @@
+                    busy.visible = false;
+                });
+
+                job.start();
+            }
+        }
+
+        Component.onCompleted: {
+            portalItem.load();
+        }
+    }
+
+    Rectangle {
+        id: buttonBackground
+        anchors {
+            left: parent.left
+            top: parent.top
+            margins: 3
+        }
+        width: childrenRect.width
+        height: childrenRect.height
+        color: "#000000"
+        opacity: .8
+        radius: 5
+
+        // catch mouse signals from propagating to parent
+        MouseArea {
+            anchors.fill: parent
+            onClicked: mouse.accepted = true
+            onWheel: wheel.accepted = true
+        }
+
+        ColumnLayout {
+            Button {
+                id: onlineMapButton
+                Layout.fillWidth: true
+                Layout.margins: 2
+                Layout.alignment: Qt.AlignHCenter
+                text: qsTr("Show Online Map")
+                enabled: !busy.visible & !viewingOnlineMaps
+                onClicked: {
+
+                    mapView.map = onlineMap;
+                    graphicsOverlay.visible = true;
+                    viewingOnlineMaps = true;
+                    mapView.setViewpointGeometry(offlineMapTask.preplannedMapAreaList.get(preplannedCombo.currentIndex).areaOfInterest);
+                    checkIfFileExists(preplannedCombo.currentIndex);
+                }
+            }
+
+            Text {
+                text: qsTr("Available Preplanned Areas:")
+                color: "white"
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 2
+            }
+
+            ComboBox {
+                id: preplannedCombo
+                Layout.fillWidth: true
+                Layout.margins: 2
+                enabled: !busy.visible
+                model: null
+                textRole: "itemTitle"
+
+                onActivated: {
+                    if (offlineMapTask.preplannedMapAreaList.count <= 0)
+                        return;
+
+                    if (offlineMapTask.preplannedMapAreaList.get(currentIndex).loadStatus !== Enums.LoadStatusLoaded)
+                        return;
+
+                    checkIfFileExists(currentIndex);
+
+                    if (viewingOnlineMaps)
+                        mapView.setViewpointGeometry(offlineMapTask.preplannedMapAreaList.get(currentIndex).areaOfInterest);
+                }
+            }
+
+            Button {
+                id: downloadOrView
+                Layout.fillWidth: true
+                Layout.margins: 2
+                enabled: !busy.visible
+                text: mapExists ? qsTr("View Preplanned area") : qsTr("Download Preplanned Area")
+                onClicked: {
+                    preplannedArea = offlineMapTask.preplannedMapAreaList.get(preplannedCombo.currentIndex);
+                    offlineMapTask.createDefaultDownloadPreplannedOfflineMapParameters(preplannedArea);
```

Comment:
I think more code comments could be added throughout. There are quite a few tasks kicked off and they are a bit buried around other app logic code, so it is hard to pick out the important code